src/main/java/com/monumental/ExceptionHandler.java

```diff
@@ -0,0 +1,27 @@
+package com.monumental;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This class is responsible for intercepting all exceptions and deciding what should happen with the HTTP response
+ * It also is able to log these exceptions, which is really helpful for debugging since Spring doesn't always print
+ * stacktraces for most exceptions
+ */
+@ControllerAdvice
```

Comment:
What does '@ControllerAdvice' do?

Reply:
I think it just means it gets applied to all the controllers. It's the standard way to do exception handling in Spring
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/ControllerAdvice.html